src/Tests/Backend/ONE/ONEToolchain.test.ts

```diff
@@ -31,13 +58,91 @@ suite('Backend', function() {
       });
     });
 
-    // suite('#getToolchainTypes', function() {
+    suite('#getToolchainTypes', function() {
+      test('returns ToolchainCompiler\'s type', function() {
+        const toolchainCompiler = new ToolchainCompiler();
+        const toolchainTypes = toolchainCompiler.getToolchainTypes();
+        assert.strictEqual(toolchainTypes.length, 1);
+        assert.strictEqual(toolchainTypes[0], 'latest');
+      });
+    });
 
-    // });
+    suite('#getVersion', function() {
+      test('returns Version object from string version', function() {
+        const toolchainCompiler = new ToolchainCompiler();
+        const version1 = toolchainCompiler.getVersion('1.0.2~RC0');
+        const version2 = new Version(1, 0, 2, '~RC0');
+        assert.isTrue(version1.equals(version2));
+        assert.isTrue(version2.equals(version1));
+      });
+      test('returns Version object from string version only with major', function() {
+        const toolchainCompiler = new ToolchainCompiler();
+        const version1 = toolchainCompiler.getVersion('1');
+        const version2 = new Version(1, 0, 0);
+        assert.isTrue(version1.equals(version2));
+        assert.isTrue(version2.equals(version1));
+      });
+      test('returns Version object from string version without patch and option', function() {
+        const toolchainCompiler = new ToolchainCompiler();
+        const version1 = toolchainCompiler.getVersion('1.0');
+        const version2 = new Version(1, 0, 0);
+        assert.isTrue(version1.equals(version2));
+        assert.isTrue(version2.equals(version1));
+      });
+      test('returns Version object from string version without option', function() {
+        const toolchainCompiler = new ToolchainCompiler();
+        const version1 = toolchainCompiler.getVersion('1.0.2');
+        const version2 = new Version(1, 0, 2);
+        assert.isTrue(version1.equals(version2));
+        assert.isTrue(version2.equals(version1));
+      });
+      test('returns Version object from string version without minor', function() {
+        const toolchainCompiler = new ToolchainCompiler();
+        const version1 = toolchainCompiler.getVersion('1.0-beta');
+        const version2 = new Version(1, 0, 0, '-beta');
+        assert.isTrue(version1.equals(version2));
+        assert.isTrue(version2.equals(version1));
+      });
+    });
 
-    // suite('#getToolchains', function() {
+    suite('#getToolchains', function() {
+      // setup(function() {
```

Comment:
Is it enough to test that the prerequisitesForGetToolchains() test produces the expected output (command string)?

Even if the above test is sufficient, it seems that you should actually register and test the ppa first to check whether getToolchains works.
I think I need to run sudo to add ppa. Is it possible to run sudo in a test environment?

Reply:
No, we can't. It is not possible to execute some command with root privilege in test environment. So it is enough to compare the command line string.